csrc/sign.cpp

```diff
@@ -452,8 +452,25 @@ JNIEXPORT jbyteArray JNICALL Java_com_amazon_corretto_crypto_provider_EvpSignatu
             paddingType, reinterpret_cast<const EVP_MD*>(mgfMdPtr), pssSaltLen);
 
         std::vector<uint8_t, SecureAlloc<uint8_t> > signature;
-        {
-            size_t sigLength;
+        size_t sigLength;
+
+        int keyType = EVP_PKEY_id(reinterpret_cast<EVP_PKEY*>(pKey));
+
+        if (keyType == EVP_PKEY_ED25519) {
+            jni_borrow message(env, messageBuf, "message");
+
+            if (!EVP_DigestSign(ctx.getDigestCtx(), NULL, &sigLength, message.data(), message.len())) {
```

Comment:
type-specific branch - so you know the output length.

If you think it's more memory safe to get it at runtime. Can just use `EVP_DigestSignFinal(ctx.getDigestCtx(), NULL, &sigLength)` and save some logic.

Reply:
EVP_DigestSignFinal does a [check in the beginning](https://github.com/aws/aws-lc/blob/0c52b12898c96f7de61933db31d7f1f681b10691/crypto/fipsmodule/evp/digestsign.c#L193-L196) that I believe would throw an error for Ed25519.

Reply:
you are quite right about that! doesn't work